### PR TITLE
[EN-17473] Propagate the variation name to the AssignmentLogData

### PR DESCRIPTION
**Pull request by simondale00**
https://glossgenius.atlassian.net/browse/EN-17473
* Propagate the variation name to the assignment log data object
* Bump the version

### CODE REVIEW

File: src/main/java/com/eppo/sdk/dto/AssignmentLogData.java [hunk 1]
```diff
@@ -29,7 +32,7 @@ public AssignmentLogData(
         this.featureFlag = featureFlag;
         this.allocation = allocation;
         this.variationValue = variationValue;
-        this.variation = variationValue.stringValue();
+        this.variation = variationName;
```
Comment (nt3rp) [inline src/main/java/com/eppo/sdk/dto/AssignmentLogData.java:35]: Comment (blocking): did we accidentally blow-away the `variation`?

I think we want this:

```suggestion
        this.variation = variationValue.stringValue();
        this.variationName = variationName;
```

File: src/main/java/com/eppo/sdk/dto/Variation.java [hunk 2]
```diff
@@ -1,12 +1,14 @@
 package com.eppo.sdk.dto;
 
 import lombok.Data;
+import lombok.EqualsAndHashCode;
```
Comment (nt3rp) [inline src/main/java/com/eppo/sdk/dto/Variation.java:4]: Comment (non-blocking): This appears to be unused ;)

**Review by nt3rp — APPROVED** [review]

**Review by nicoraynaud — APPROVED** [review]
LGTM ! (with nick's suggestion on the variation & variation name)

File: src/main/java/com/eppo/sdk/dto/Variation.java (hunk 2 above)
Comment (simondale00) [reply src/main/java/com/eppo/sdk/dto/Variation.java:4]: Ha! Ooops.

File: src/main/java/com/eppo/sdk/dto/AssignmentLogData.java (hunk 1 above)
Comment (simondale00) [reply src/main/java/com/eppo/sdk/dto/AssignmentLogData.java:35]: We have the `variationValue`, which is an EppoValue (typed).

Did we still want the `variation` to be a string representation of the eppo value?

File: src/main/java/com/eppo/sdk/dto/Variation.java (hunk 2 above)
Comment (simondale00) [reply src/main/java/com/eppo/sdk/dto/Variation.java]: Addressed in `b6fd0cb`.

File: src/main/java/com/eppo/sdk/dto/AssignmentLogData.java (hunk 1 above)
Comment (simondale00) [reply src/main/java/com/eppo/sdk/dto/AssignmentLogData.java]: Spoke offline. `this.variation = variationValue.stringValue()` was part of Eppo's initial implementation. To maintain backwards compatibility we're going to leave that property as-is.

As such, I'm going to add a `variationName` property to capture the name of the variation.

**Pull request merged**